Science 2022 - Polling Sensors - Temperature and RH

Stanlist opened the pull request:
This PR sets up the driver code for the DHT22 Temperature and Relative Humidity sensor. The drivers for the sensor are adapted from https://os.mbed.com/users/Wimpie/code/DHT/. This is still a draft as the code still needs to be tested.

jahnavithota2011 commented:
@Stanlist I few questions regarding your PR:
- Did you build your `DHT22` files of https://os.mbed.com/users/Wimpie/code/DHT//file/9b5b3200688f/DHT.cpp/. If yes, is there a reason why you are not using the original files.
- When you were testing the sensors, were you able to get sensible values? If yes how stable were the readings.
- Your PR has merge conflicts, can you please resolve and rebase the branch.

Stanlist commented:
The code I used it taken from the DHT.cpp library, I just made some changes to it to work with our module system. I removed some of the code that wouldn't be used in our system. 

When I was testing, I was able to get integer values for the temperature and RH readings from the sensor, but I noticed that it would often fail the checksum that ensures the data read is valid. Other than that, the data I got looked accurate and changed appropriately when I tried grasping it in my hand or blowing on it.
